src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
+				{
+					IBakedModel partModel = modelLookup.apply( part );
+					QuadRotator rotator = new QuadRotator();
+					quads.addAll( rotator.rotateQuads( gatherQuads( partModel, null, rand ), side, EnumFacing.UP ) );
+				}
 			}
-			catch( Throwable t )
+			//If we are forcing transparency and this isn't the Translucent layer.
+			if( transparent && layer != BlockRenderLayer.TRANSLUCENT )
 			{
-				AELog.debug( t );
+				continue;
 			}
-		} );
-	}
-
-	public static TextureAtlasAndTint getSprite( IBakedModel blockModel, IBlockState state, EnumFacing facing, long rand )
-	{
 
-		TextureAtlasAndTint firstFound = null;
-		BlockRenderLayer orgLayer = MinecraftForgeClient.getRenderLayer();
-
-		try
-		{
-			// Some other mods also distinguish between layers, so we're doing this in a loop from most likely to least
-			// likely
-			for( BlockRenderLayer layer : BlockRenderLayer.values() )
+			IBlockState blockState = facadeRenderState.getSourceBlock();
+			//If we aren't forcing transparency let the block decide if it should render.
+			if( !transparent )
 			{
-
-				ForgeHooksClient.setRenderLayer( layer );
-
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, facing, rand ) )
+				if( !blockState.getBlock().canRenderInLayer( blockState, layer ) )
 				{
-					return new TextureAtlasAndTint( bakedQuad );
+					continue;
 				}
+			}
 
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, null, rand ) )
+			AxisAlignedBB fullBounds = thinFacades ? THIN_FACADE_BOXES[sideIndex] : THICK_FACADE_BOXES[sideIndex];
+			AxisAlignedBB facadeBox = fullBounds;
+			//If we are a transparent facade, we need to modify out BB.
+			if( facadeRenderState.isTransparent() )
+			{
+				double offset = thinFacades ? THIN_THICKNESS : THICK_THICKNESS;
+				AEAxisAlignedBB tmpBB = null;
+				for( EnumFacing face : EnumFacing.VALUES )
 				{
-					if( firstFound == null )
-					{
-						firstFound = new TextureAtlasAndTint( bakedQuad );
-					}
-					if( bakedQuad.getFace() == facing )
+					//Only faces that aren't on our axis
+					if( face.getAxis() != side.getAxis() )
 					{
-						return new TextureAtlasAndTint( bakedQuad );
+						FacadeRenderState otherState = facadeStates.get( face );
+						if( otherState != null && !otherState.isTransparent() )
+						{
+							if( tmpBB == null )
+							{
+								tmpBB = AEAxisAlignedBB.fromBounds( facadeBox );
+							}
+							switch( face )
+							{
+								case DOWN:
+									tmpBB.minY += offset;
+									break;
+								case UP:
+									tmpBB.maxY -= offset;
+									break;
+								case NORTH:
+									tmpBB.minZ += offset;
+									break;
+								case SOUTH:
+									tmpBB.maxZ -= offset;
+									break;
+								case WEST:
+									tmpBB.minX += offset;
+									break;
+								case EAST:
+									tmpBB.maxX -= offset;
+									break;
+								default:
+									throw new RuntimeException( "Switch falloff. " + String.valueOf( face ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Directly append the argument of String.valueOf(). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1153)